client/Hooks/useLogger/useLogger.steps.tsx

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+import { And, Given, When, Then, Fusion, After } from 'jest-cucumber-fusion';
+import { renderHook } from '@testing-library/react-hooks';
+import { Server } from 'mock-socket';
+
+import { useLogger, LoggerType, MESSAGE_BUFFER_MAX_SIZE } from '.';
+
+let logger: LoggerType;
+let rerenderHook: (newProps?: unknown) => void;
+let mockWebSocketServer: Server;
+let onConnectionPromise: Promise<void>;
+let onMessagePromise: Promise<string>;
+let sentMessagesCount = 0;
+let verifiedClient = true;
+
+After(() => {
+  sentMessagesCount = 0;
+  mockWebSocketServer.close();
+  verifiedClient = true;
+});
+
+Given('a logging WebSocket server', () => {
```

Comment:
Worry is that if jest runs tests in parallel, or someone else uses this `Given` we'll get port clashes. But if this is fixed via the context change as a later commit (or new PR) I'm less worried. Wonder if when you add in the `context` we can supply a mock ws client via context instead of having to start up real servers in the test.

Reply:
The `new Server(..)` line here is a [`mock-socket`](https://www.npmjs.com/package/mock-socket) server, which just hooks mock-clients to mock-servers using the supplied server URL as a key. No real servers are being started here :)